### ISSUE TITLE
RTL: offset-*, pull-* and push-* not rendering properly

### ISSUE THREAD

**Issue by abennouna** [ionic-team/ionic#16701]
# Bug Report

**Ionic version:**
[x] **4.x**

**Current behavior:**
When the document is RTL, the `offset-*`, `push-*` and `pull-*` directives do not render as expected.

**Expected behavior:**
The offset, pull, and push, should work on RTL.

**Steps to reproduce:**
1. Go to the [demo](https://abennouna.github.io);
2. If your display is already LTR, skip to 4;
3. Click on the "TOGGLE LTR / RTL" button in order to get a LTR display;
4. Scroll down a bit to the grids, and observe the correct offsets, pulls, and pushs;
5. Click on the "TOGGLE LTR / RTL" button in order to get an RTL display;
6. Notice the incorrect offsets, pulls, and pushs.

**Related code:**
- Source: https://github.com/abennouna/ionic4-rtl-issue
- Demo: https://abennouna.github.io

**Other information:**
Currently, the `Col` component hard-codes the `margin-left`, `left`, and `right` CSS properties.

I'm creating a PR for this.

**Ionic info:** 
```
Ionic:

   ionic (Ionic CLI)             : 4.5.0 (~/.nvm/versions/node/v10.12.0/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.0.0-beta.17
   @angular-devkit/build-angular : 0.10.7
   @angular-devkit/schematics    : 7.0.7
   @angular/cli                  : 7.0.7
   @ionic/angular-toolkit        : 1.2.0

System:

   NodeJS : v10.12.0 (~/.nvm/versions/node/v10.12.0/bin/node)
   npm    : 6.4.1
   OS     : macOS High Sierra
```

**Comment by brandyscarney** [comment]
Thank you for the PR! We will review this.

**Comment by mohamedsharaf** [comment]
hope this was fixed b4 the release 0 @brandyscarney

**Comment by abennouna** [comment]
Closed in favor of #17012
